src/types.rs

```diff
@@ -264,87 +264,63 @@ fn map_hex_string<T: Hex>(v: &Option<T>) -> Option<String> {
     v.as_ref().map(|v| v.encode_hex())
 }
 
-pub(crate) fn map_opt_result<T, V, F>(v: Option<T>, f: F) -> Result<Option<V>>
+pub(crate) fn map_opt_result<T, V, F, E>(v: Option<T>, f: F) -> std::result::Result<Option<V>, E>
 where
-    F: Fn(T) -> Result<V>,
+    F: Fn(T) -> std::result::Result<V, E>,
 {
     match v {
         Some(v) => Ok(Some(f(v)?)),
         None => Ok(None),
     }
 }
 
-fn map_i64_from_u64(v: Option<u64>) -> Result<Option<i64>> {
-    map_opt_result(v, |v| i64::try_from(v).context("converting u64 to i64"))
-}
-
-fn map_i64<T: AsRef<[u8]>>(v: &Option<T>) -> Result<Option<i64>> {
-    map_opt_result(v.as_ref(), |v| {
-        v.as_ref()
-            .try_into()
-            .context("converting to fixed size bytes i64")
-            .map(i64::from_be_bytes)
+fn map_i64<T: AsRef<[u8]>>(opt: &Option<T>) -> Result<Option<i64>> {
+    map_opt_result(opt.as_ref(), |v| {
+        i64::try_from(ruint::aliases::U256::from_be_slice(v.as_ref()))
+            .context("converting U256 to i64")
```

Comment:
@ozgrakkurt the only issue with this is it cannot be used for negative number conversions. I don't think cases we are using it for have any negative numbers though so this should be fine.

Reply:
If it is negative you can use signed 256bit integer from ruint and use convert_bigint_signed function after